File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.js
function solution(numbers) {
    const initCounts = new Array(101).fill(0);
    const retCounts = new Array(201).fill(0);
    let count = 0;
    for (let i = 0; i < numbers.length; i++) {
        if (initCounts[numbers[i]]++ == 1 && retCounts[numbers[i] * 2] < 1) {
            retCounts[numbers[i] + numbers[i]]++;
            count++;
        }
        else if (initCounts[numbers[i]] == 1) {
            for (let j = i - 1; j >= 0; j--) {
                if (retCounts[numbers[i] + numbers[j]]++ == 0) count++;
            }
        }
    }
    let ret = new Array(count);
    let ind = 0;
    for (let i = 0; i < retCounts.length; i++) {
        if (retCounts[i] > 0) ret[ind++] = i;
    }
    return ret;
}

File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.java
class Solution {
    public int[] solution(int[] numbers) {
        boolean[] initCount = new boolean[101];
        boolean[] retCount = new boolean[201];
        int count = 0;
        for (int i = 0; i < numbers.length; i++) {
            if (initCount[numbers[i]] && !retCount[numbers[i] * 2]) {
                retCount[numbers[i] * 2] = true;
                count++;
                continue;
            }
            initCount[numbers[i]] = true;
            for (int j = i - 1; j >= 0; j--) {
                if (!retCount[numbers[i] + numbers[j]]) {
                    retCount[numbers[i] + numbers[j]] = true;
                    count++;
                }
            }
        }
        int[] ret = new int[count];
        int ind = 0;
        for (int i = 0; i < retCount.length; i++) {
            if (retCount[i]) ret[ind++] = i;
        }
        
        return ret;
    }
}